packages/core/test/app/utils/number.spec.ts

```diff
@@ -0,0 +1,320 @@
+
+    it('test negatives', () => {
+      assert(new BigNumber(-10).gt(new BigNumber(-11)))
+      assert(!new BigNumber(-11).gt(new BigNumber(-10)))
+      assert(!new BigNumber(-10).gt(new BigNumber(-10)))
+    })
+  })
+
+  describe('gte', () => {
+    it('test positive', () => {
+      assert(new BigNumber(10).gte(new BigNumber(9)))
+      assert(!new BigNumber(9).gte(new BigNumber(10)))
+      assert(new BigNumber(10).gte(new BigNumber(10)))
+    })
+
+    it('test negative', () => {
+      assert(new BigNumber(10).gte(new BigNumber(-11)))
+      assert(!new BigNumber(-11).gte(new BigNumber(10)))
+    })
+
+    it('test negatives', () => {
+      assert(new BigNumber(-10).gte(new BigNumber(-11)))
+      assert(!new BigNumber(-11).gte(new BigNumber(-10)))
+      assert(new BigNumber(-10).gte(new BigNumber(-10)))
+    })
+  })
+
+  describe('lt', () => {
+    it('test positive', () => {
+      assert(!new BigNumber(10).lt(new BigNumber(9)))
+      assert(new BigNumber(9).lt(new BigNumber(10)))
+      assert(!new BigNumber(10).lt(new BigNumber(10)))
+    })
+
+    it('test negative', () => {
+      assert(!new BigNumber(10).lt(new BigNumber(-11)))
+      assert(new BigNumber(-11).lt(new BigNumber(10)))
+    })
+
+    it('test negatives', () => {
+      assert(!new BigNumber(-10).lt(new BigNumber(-11)))
+      assert(new BigNumber(-11).lt(new BigNumber(-10)))
+      assert(!new BigNumber(-10).lt(new BigNumber(-10)))
+    })
+  })
+
+  describe('lte', () => {
+    it('test positive', () => {
+      assert(!new BigNumber(10).lte(new BigNumber(9)))
+      assert(new BigNumber(9).lte(new BigNumber(10)))
+      assert(new BigNumber(10).lte(new BigNumber(10)))
+    })
+
+    it('test negative', () => {
+      assert(!new BigNumber(10).lte(new BigNumber(-11)))
+      assert(new BigNumber(-11).lte(new BigNumber(10)))
+    })
+
+    it('test negatives', () => {
+      assert(!new BigNumber(-10).lte(new BigNumber(-11)))
+      assert(new BigNumber(-11).lte(new BigNumber(-10)))
+      assert(new BigNumber(-10).lte(new BigNumber(-10)))
+    })
+  })
+
+  describe('eq', () => {
+    it('test positive', () => {
+      assert(new BigNumber(10).eq(new BigNumber(10)))
+      assert(!new BigNumber(9).eq(new BigNumber(10)))
+    })
+
+    it('test negative', () => {
+      assert(new BigNumber(-10).eq(new BigNumber(-10)))
+      assert(!new BigNumber(-11).eq(new BigNumber(-10)))
+    })
+  })
+
+  describe('compare', () => {
+    it('test first is less', () => {
+      assert(new BigNumber(10).compare(new BigNumber(11)) === -1)
+      assert(new BigNumber(-11).compare(new BigNumber(-10)) === -1)
+    })
+
+    it('test second is less', () => {
+      assert(new BigNumber(11).compare(new BigNumber(10)) === 1)
+      assert(new BigNumber(-10).compare(new BigNumber(-11)) === 1)
+    })
+
+    it('test equal', () => {
+      assert(new BigNumber(10).compare(new BigNumber(10)) === 0)
+      assert(new BigNumber(-10).compare(new BigNumber(-10)) === 0)
+    })
+  })
+})
```

Comment:
Considering how comprehensive these tests are it is worth noting that some `toX()` functions were not tested. In particular toBuffer, toJSON, & toString().